ingest-mapper/src/main/scala/uk/gov/nationalarchives/ingestmapper/Lambda.scala

```diff
@@ -56,12 +59,22 @@ class Lambda extends LambdaRunner[Input, StateOutput, Config, Dependencies] {
       _ <- log(s"Processing batchRef ${input.batchId}")
 
       discoveryService <- dependencies.discoveryService
-      departmentAndSeries <- discoveryService.getDepartmentAndSeriesRows(input)
+      hundredDaysFromNowInEpochSecs <- IO {
+        val hundredDaysFromNow: Instant = dependencies.time().plus(100, ChronoUnit.DAYS)
+        Num(hundredDaysFromNow.getEpochSecond.toDouble)
+      }
+      departmentAndSeries <- discoveryService.getDepartmentAndSeriesItems(input, hundredDaysFromNowInEpochSecs)
       _ <- log(s"Retrieved department and series ${departmentAndSeries.show}")
 
       bagManifests <- dependencies.metadataService.parseBagManifest(input)
       bagInfoJson <- dependencies.metadataService.parseBagInfoJson(input)
-      metadataJson <- dependencies.metadataService.parseMetadataJson(input, departmentAndSeries, bagManifests, bagInfoJson.headOption.getOrElse(Obj()))
+      metadataJson <- dependencies.metadataService.parseMetadataJson(
+        input,
+        departmentAndSeries,
+        bagManifests,
+        bagInfoJson.headOption.getOrElse(Obj()),
+        hundredDaysFromNowInEpochSecs
```

Comment:
Instead of having to pass what is essentially a constant into two different methods, would it be easier to add the ttl attribute in the line below.
Something like
```scala
metadataJsonWithTtl <- IO(metadataJson.map(obj => Obj.from(obj.value ++ Map("ttl" -> hundredDaysFromNowInEpochSecs))))
```
This way, you guarantee you've got all the items as it's just before the write to dynamo and you don't have to pass around the hundred years parameter. 
It also means you're not testing essentially the same thing in each test class.

You'd have to re-jig the tests a bit, I got it running on a branch though.  https://github.com/nationalarchives/dr2-ingest/compare/DR2-1706_addTtlAttributeToAllItemsInIngestMapper...suggested-changes?expand=1